src/main/java/seedu/duke/Duke.java

```diff
@@ -30,18 +28,21 @@ public Duke() {
      */
     public static void main(String[] args) throws IOException {
         new Duke();
+
         favFile.readFile();
-        boolean isOngoing = true;
         try {
             freqFile.readFile();
         } catch (CustomException e) {
             Ui.showError(e);
         }
+
         Ui.printWelcomeMessage();
+        parser = new Parser("lol");
+        boolean isOngoing = true;
         while (isOngoing) {
             try {
                 String fullCommand = Ui.getCommand();
-                parser = new Parser(fullCommand);
+                parser.setUserInput(fullCommand);
```

Comment:
Why use setUserInput instead of making new Parser object?

Reply:
because I need the previous input to save into fav, new parser cannot save the old input